app/src/main/java/com/provider/service/ProviderValidator.java

```diff
@@ -0,0 +1,39 @@
+package com.provider.service;
+
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import com.provider.exception.BadRequestException;
+import com.provider.model.AccountRoleIDReturnModel;
+import com.provider.model.ProviderRequestModel;
+import com.provider.persistence.repository.ProviderRepository;
+
+import lombok.RequiredArgsConstructor;
+
+@Service
+@RequiredArgsConstructor
+public class ProviderValidator {
+
+    private final RestTemplate restTemplate;
+
+    private final ProviderRepository providerRepository;
+
+    public void validateProviderRequest(UUID accountID, ProviderRequestModel providerRequestModel) {
+        String url="http://localhost:3000/api/v1/account/" + accountID.toString() + "/role/PROVIDER";
```

Comment:
this piece `http://localhost:3000` of URL needs to be configurable. Let's put this base account API URL as property in `application.properties`

Reply:
Also lets use URI templates https://docs.spring.io/spring-framework/reference/integration/rest-clients.html#rest-resttemplate-uri